### PR TITLE
get var by experiment

### PR DESCRIPTION
**Pull request by JeanEudesH**
function to get all variables linked to an experiment.
Plus the get data function to get data from those variables

### CODE REVIEW

File: functions.py [hunk 1]
```diff
@@ -482,3 +482,106 @@ def add_data_from(pythonClient,data_csv):
         if "DUPLICATE" in str(e):
             logging.error("Duplicate data")
         exit()
+
+
+def getVariablesbyURI(uri="", pythonClient=""):
```
Comment (niio972) [inline functions.py]: `def getVariablesbyURI(uri="", pythonClient=None):`

Comment (niio972) [reply functions.py]: It's better to use None instead an empty character variable to define these default values

File: functions.py [hunk 2]
```diff
@@ -482,3 +482,106 @@ def add_data_from(pythonClient,data_csv):
...
+        variables_details = pd.DataFrame()
+        for variables in var_in_expe['result']:
+            variables_details = variables_details.append(getVariablesbyURI(uri=[variables.uri], pythonClient = pythonClient))
+        return variables_details
+    except Exception as e:
+        print("Exception : %s\n" % e)
+        return None
+
+
+def get_data(experiment=[], variables=[], scientific_objects=[], pythonClient="", start_date=[], end_date=[]):
```
Comment (niio972) [inline functions.py]: Use None instead of "" for pythonClient variable

Comment (niio972) [inline functions.py]: You may use `None` for `start_date` or `end_date` default values

File: functions.py [hunk 3]
```diff
@@ -482,3 +482,106 @@ def add_data_from(pythonClient,data_csv):
...
+            variable_list = {"name": Name, "Uri": Uri, "alternative_name": Alternative_name, "entity_name": Entity_name, "entity_uri": Entity_uri, "characteristic_name": Characteristic_name,
+                             "characteristic_uri": Characteristic_uri, "method_name": Method_name, "method_uri": Method_uri, "unit_name": Unit_name, "unit_uri": Unit_uri, "description": Description}
+            var_data = pd.DataFrame(variable_list)
+        return var_data
+    except Exception as e:
+        print("Exception : %s\n" % e)
+        return None
+
+
+def getVariablesByExperiment(experiment="", name="", year="", species="", factors="", pythonClient=""):
```
Comment (niio972) [inline functions.py]: It's better to use None instead an empty character variable to define complex object default values

**Review by niio972 — COMMENTED** [review]
Where can I find some examples

**Comment by JeanEudesH** [comment]
For testing 

pythonClient = opensilexClientToolsPython.ApiClient()
pythonClient.connect_to_opensilex_ws(
    identifier="guest@opensilex.org", password="guest", host="http://138.102.159.36:8084/rest")

var_frai = getVariablesByExperiment(experiment="test-expe:frai21", pythonClient= pythonClient)
var_Demo = getVariablesByExperiment(experiment="test-expe:demo_systeme", pythonClient= pythonClient)

File: functions.py (hunk 1 above)
Comment (JeanEudesH) [reply functions.py]: Corrected in [commit](https://github.com/OpenSILEX/opensilexClientPython/pull/1/commits/d4265714dde879b5f17d55e913a55ab232b393d6)

File: functions.py (hunk 2 above)
Comment (JeanEudesH) [reply functions.py]: Corrected in [commit](https://github.com/OpenSILEX/opensilexClientPython/pull/1/commits/d4265714dde879b5f17d55e913a55ab232b393d6)

Comment (JeanEudesH) [reply functions.py]: Corrected in [commit](https://github.com/OpenSILEX/opensilexClientPython/pull/1/commits/d4265714dde879b5f17d55e913a55ab232b393d6)

**Pull request closed without merging**